### PR TITLE
remove warnings from data models file uploads

### PR DESCRIPTION
**Pull request by toondaey**

### CODE REVIEW

**Review by einarmo — COMMENTED** [review]
The PR title does not match the content of the PR

File: cognite/extractorutils/configtools/elements.py [hunk 1]
```diff
@@ -696,3 +696,37 @@ def create_state_store(
             return LocalStateStore(file_path="states.json", cancellation_token=cancellation_token)
         else:
             return NoStateStore()
+
+
+class RegExpFlag(Enum):
+    IGNORECASE = "i"
+    ASCII_ONLY = "a"
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:703]: I really don't like making users configure fields with somewhat mystical `i` or `a` flags.

We should be using more human-readable fields

File: cognite/extractorutils/configtools/elements.py [hunk 2]
```diff
@@ -696,3 +696,37 @@ def create_state_store(
...
+
+
+@dataclass
+class IgnorePattern:
+    """
+    Configuration for regexp for ignore pattern
+    """
+
+    pattern: str
+    flags: list[RegExpFlag]
```
Comment (mathialo) [inline cognite/extractorutils/configtools/elements.py:720]: I'm also not convinced that 'flags' is the best name for this. Options perhaps?

File: cognite/extractorutils/uploader/files.py [hunk 3]
```diff
@@ -296,7 +296,6 @@ def _create_multi_part(self, file_meta: FileMetadataOrCogniteExtractorFile, chun
         if isinstance(file_meta, CogniteExtractorFileApply):
             node_id = self._apply_cognite_file(file_meta)
             identifiers = IdentifierSequence.load(instance_ids=node_id).as_singleton()
-            self.cdf_client.files._warn_alpha()
```
Comment (mathialo) [inline cognite/extractorutils/uploader/files.py:299]: These two things are quite unrelated. It should really be two different PRs.

File: cognite/extractorutils/configtools/loaders.py [hunk 4]
```diff
@@ -315,6 +321,57 @@ def load_yaml_dict(
...
+    Check string against list of RegExp patterns.
+
+    Args:
+        patterns: A list of (re) patterns to match string against.
+        string: String to which we match the pattern.
+
+    Returns:
+        boolean value indicating whether string matches any of the patterns.
+    """
+    return True in [matches_pattern(pattern, string) for pattern in patterns]
```
Comment (mathialo) [inline cognite/extractorutils/configtools/loaders.py:335]:
``` suggestion
    return any([matches_pattern(pattern, string) for pattern in patterns])
```

File: cognite/extractorutils/configtools/loaders.py [hunk 5]
```diff
@@ -315,6 +321,57 @@ def load_yaml_dict(
     )
 
 
+def matches_patterns(patterns: list[Union[str, re.Pattern[str]]], string: str) -> bool:
```
Comment (mathialo) [inline cognite/extractorutils/configtools/loaders.py:324]: These don't really belong in the `loaders` module IMO.

**Review by mathialo — APPROVED** [review]
